src/internal/sagaHelpers.js

```diff
@@ -21,7 +21,7 @@ function fsmIterator(fsm, q0, name = 'iterator') {
       let [q, output, _updateState] = fsm[qNext]()
       qNext = q
       updateState = _updateState
-      return qNext === qEnd ? done : output
+      return qNext === qEnd || output === END ? done : output
```

Comment:
I think that the END test is not required anymore  b/c `proc` will anyway terminate the generator if it detects an END (END is returned explicitly only when using `takem` effects ). see

https://github.com/yelouafi/redux-saga/blob/master/src/internal/proc.js#L412
https://github.com/yelouafi/redux-saga/blob/master/src/internal/proc.js#L247-L250

I added this test when there wasn't yet automatic handling of END in proc but forgot to remove it later.